MekHQ/src/mekhq/gui/dialog/CampaignOptionsDialog.java

```diff
@@ -4609,13 +4354,434 @@ private void fillRankInfo() {
+                if (eras.size() > 0) {
+                    displayName.append(" (").append(eras.get(0));
+                    if (eras.size() > 1) {
+                        displayName.append("-").append(eras.get(eras.size() - 1));
+                    }
+                    displayName.append(")");
+                }
+                if (availableRatModel.contains(displayName.toString())) {
+                    chosenRatModel.addElement(displayName.toString());
+                    availableRatModel.removeElement(displayName.toString());
+                }
+            }
+        }
+        chkIgnoreRatEra.setSelected(options.canIgnoreRatEra());
+
+        chkDoubleVehicles.setSelected(options.getDoubleVehicles());
+        spnOpforLanceTypeMechs.setValue(options.getOpforLanceTypeMechs());
+        spnOpforLanceTypeMixed.setValue(options.getOpforLanceTypeMixed());
+        spnOpforLanceTypeVehicles.setValue(options.getOpforLanceTypeVehicles());
+        chkOpforUsesVTOLs.setSelected(options.getOpforUsesVTOLs());
+        chkOpforUsesAero.setSelected(options.getAllowOpforAeros());
+        spnOpforAeroChance.setValue(options.getOpforAeroChance());
+        chkOpforUsesLocalForces.setSelected(options.getAllowOpforLocalUnits());
+        spnOpforLocalForceChance.setValue(options.getOpforLocalUnitChance());
+        chkAdjustPlayerVehicles.setSelected(options.getAdjustPlayerVehicles());
+        chkRegionalMechVariations.setSelected(options.getRegionalMechVariations());
+        chkAttachedPlayerCamouflage.setSelected(options.getAttachedPlayerCamouflage());
+        chkPlayerControlsAttachedUnits.setSelected(options.getPlayerControlsAttachedUnits());
+        chkUseDropShips.setSelected(options.getUseDropShips());
+        chkUseWeatherConditions.setSelected(options.getUseWeatherConditions());
+        chkUseLightConditions.setSelected(options.getUseLightConditions());
+        chkUsePlanetaryConditions.setSelected(options.getUsePlanetaryConditions());
+        chkUseAtBCapture.setSelected(options.getUseAtBCapture());
+        spnStartGameDelay.setValue(options.getStartGameDelay());
+        //endregion Against the Bot Tab
+
+        //region Miscellaneous Tab
+        chkHistoricalDailyLog.setSelected(options.historicalDailyLog());
+        //endregion Miscellaneous Tab
+    }
+
+    public static String[][] getSkillCostsArray(Hashtable<String, SkillType> skillHash) {
+        String[][] array = new String[SkillType.getSkillList().length][11];
+        int i = 0;
+        for (String name : SkillType.getSkillList()) {
+            SkillType type = skillHash.get(name);
+            for (int j = 0; j < 11; j++) {
+                array[i][j] = Integer.toString(type.getCost(j));
+            }
+            i++;
+        }
+        return array;
+    }
+    //endregion Initialization
+
+    @SuppressWarnings(value = "unused") // FIXME:
     private void tableRanksValueChanged(javax.swing.event.ListSelectionEvent evt) {
```

Comment:
What's the point of having this method if it does nothing and isn't used anyway?

Reply:
This is something that we should have, but was broken it seems.